### ISSUE TITLE
Investigate app/evm DB divergence after power failure

### ISSUE THREAD

**Issue by enlight** [loomnetwork/loomchain#1359]
One of the external PlasmaChain validators encountered this error after a power failure:
```
Jul 18 07:47:27 X[5806]: Error: EVM roots mismatch, evm.db(8076865): 6F753C94CA47298CE34B055F9A98217FA690618E4BC6161B1E65E6005F3E5254, app.db(8076973): 2494B5DD066018D3AAD48D0C8C966622D18F9093A387DBB5CDBDA1D805B2D7
```

We should investigate how it's possible to end up in this state, and what can be done to either prevent this from happening, or to auto-recover when it does happen.

**Comment by pathornteng** [comment]
According to MultiWriterAppStore logic, we save EVM root to evm.db before app.db. So suppose the power failure happens right after evm.db is committed with changed EVM root at height 5, and app.db is still at height 4. When a node starts, MultiWriterAppStore will load EVM root at height 4 from evm.db root and the roots should match.

Is it possible that GolevelDB or Tendermint GolevelDB loses some field?

We can implement auto-recovery mechanism by making the app try to reload EVM root from evm.db at the exact height of app.db that has EVM root saved.

**Comment by enlight** [comment]
From that error `evm.db` was at version `8076865`, and `app.db` was at version `8076973`, so `app.db` was 108 versions ahead.

**Comment by pathornteng** [comment]
> From that error `evm.db` was at version `8076865`, and `app.db` was at version `8076973`, so `app.db` was 108 versions ahead.

Right, I misread the number. So it makes sense that the app.db height is higher than evm.db but it does not make sense that app.db has a EVM root saved at higher height since evm.db must be committed before app.db. Not sure how this can happen.

To fix this, we can have to check EVM roots before we load app.db and evm.db. Similar solution to https://github.com/loomnetwork/loomchain/issues/1335
